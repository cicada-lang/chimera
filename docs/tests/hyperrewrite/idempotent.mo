// If a constraint already in the result, a hyperrule should fail.

hyperrule h1 {
  [LtEq(x, y), LtEq(y, z)] => [LtEq(x, y), LtEq(y, z), LtEq(x, z)]
}

print hyprrewriteManySteps(10, h1, quote [LtEq(x, y), LtEq(y, z)])
print h1(quote [LtEq(x, y), LtEq(y, z)])
