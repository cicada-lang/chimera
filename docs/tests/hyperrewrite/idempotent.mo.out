[LtEq(1, 3), LtEq(1, 2), LtEq(2, 3), LtEq(1, 3)]
[LtEq(1, 3), LtEq(1, 2), LtEq(2, 3), LtEq(1, 3)]
[LtEq(1, 3), LtEq(1, 2), LtEq(2, 3)]
[LtEq(1, 3), LtEq(1, 2), LtEq(2, 3), LtEq(1, 3)]
[LtEq(1, 3), LtEq(10, 20), LtEq(20, 30)]
[ 
  [LtEq(1, 2), LtEq(2, 3)],
  [LtEq(1, 2), LtEq(2, 3), LtEq(1, 3)],
  [LtEq(1, 3), LtEq(1, 2), LtEq(2, 3), LtEq(1, 3)]
]
